### PR TITLE
libtego: added definitions for file transfer API

### PR DESCRIPTION
**Pull request by pospeselr**
update to tego.h to expose file transfer functionality to the UI

### CODE REVIEW

**Comment by m-simonelli** [comment]
I'm probably misunderstanding something here, but all these functions declarations are missing definitions?

**Comment by pospeselr** [comment]
Yeah just a proposed API I'm currently wiring through

File: src/libtego/source/file_hash.cpp [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+tego_file_hash::tego_file_hash(std::istream& stream)
+: tego_file_hash()
+{
+    // init sha3 512 algo
+    std::unique_ptr<::EVP_MD_CTX> ctx(EVP_MD_CTX_new());
+    EVP_DigestInit_ex(ctx.get(), EVP_sha3_512(), nullptr);
+
+    // alloc a temp 64k buffer to read bytes into
+    constexpr size_t BLOCK_SIZE = 65536;
```
Comment (m-simonelli) [inline src/libtego/source/file_hash.cpp:48]: should this go in file_hash.hpp, along with the other crypto related constexprs?

**Comment by pospeselr** [comment]
Closing this for now in favor of single squashed commit

**Pull request closed without merging**